Applications/Spire/Source/Ui/CalendarDatePicker.cpp

```diff
@@ -46,6 +50,53 @@ namespace {
   }
 }
 
+class CalendarListModel : public ListModel {
+  public:
+    static const int DAY_COUNT = 42;
+
+    CalendarListModel(std::shared_ptr<DateModel> model)
+        : m_model(std::move(model)),
+          m_current_connection(m_model->connect_current_signal(
+            [=] (auto current) { on_current(current); })) {
+      on_current(m_model->get_current());
+    }
+
+    const std::shared_ptr<DateModel>& get_model() const {
+      return m_model;
+    }
+
+    int get_size() const override {
+      return DAY_COUNT;
+    }
+
+    const std::any& at(int index) const override {
+      return m_dates[index];
+    }
+
+    connection connect_operation_signal(
+        const OperationSignal::slot_type& slot) const override {
+      return m_transaction.connect_operation_signal(slot);
+    }
+
+  private:
+    std::shared_ptr<DateModel> m_model;
+    scoped_connection m_current_connection;
+    std::array<std::any, DAY_COUNT> m_dates;
+    ListModelTransactionLog m_transaction;
+
+    void on_current(date current) {
+      auto day = date(current.year(), current.month(), 1);
```

Comment:
Think about the meeting we had two weeks ago and incorporate those concepts into the implementation.